src/helpers/getAssetData.ts

```diff
@@ -0,0 +1,41 @@
+import { Types } from "@stellar/wallet-sdk";
+import { getAssetSettingsFromToml } from "helpers/getAssetSettingsFromToml";
+import { normalizeAssetProps } from "helpers/normalizeAssetProps";
+import { Asset } from "types/types.d";
+
+// TODO: add logs
+export const getAssetData = async ({
+  balances,
+  networkUrl,
+}: {
+  balances: Types.BalanceMap;
+  networkUrl: string;
+}) => {
+  const allAssets = Object.entries(balances);
+  const assets: {
+    [key: string]: Asset;
+  } = {};
+
+  if (!allAssets?.length) {
+    return assets;
+  }
+
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < allAssets.length; i++) {
+    const [assetId, data] = allAssets[i];
+
+    // eslint-disable-next-line no-await-in-loop
+    const { homeDomain, supportedActions } = await getAssetSettingsFromToml({
+      assetId,
+      networkUrl,
+    });
+
+    assets[assetId] = normalizeAssetProps({
+      source: data,
+      homeDomain,
+      supportedActions,
+    });
+  }
```

Comment:
ah, for loop because using `forEach` with `await` inside wasn't resolving properly, right? I just had a similar situation. Nothing wrong with your code, but in case you're curious, you can also do this:

```
await Promise.all(
  allAssets.map(async ([assetId, data]) => {
    ...await whatever you need to...
  })
  
)
```

It will wait for each Promise to resolve before moving on to the next
Here's a good explanation on it: https://stackoverflow.com/a/37576787

Reply:
We will keep for loop because we need to make sure the sequence of the assets doesn't change.